plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -165,19 +164,19 @@ def release_statuses(pods, release_doc)
 
       pods = pods.select { |pod| pod.role_id == role.id && pod.deploy_group_id == group.id }
 
-      statuses = if pods.empty?
-        [[false, "Missing"]]
-      else
-        pods.map do |pod|
-          if pod.live?
-            if pod.restarted?
-              [false, RESTARTED]
-            else
-              [true, "Live"]
-            end
+      statuses = Array.new(release_doc.desired_pod_count).each_with_index.map do |_, i|
```

Comment:
marking pods missing here now then waiting on the outside ... should be more consistent ...